cli/tests/testdata/worker_terminate_doesnt_block.js

```diff
@@ -0,0 +1,20 @@
+const WORKER_CODE = "while (true) {}";
+
+const worker = new Worker(
+  `data:text/javascript,${WORKER_CODE}`,
+  { type: "module" },
+);
+
+const before = Date.now();
+worker.terminate();
+const after = Date.now();
+
+// Terminating a worker that's stuck in an infinite loop will take 2 seconds.
+// Let's take .5 seconds as definitely indicating a bug.
+if (after - before >= 500) {
+  console.log(
+    `Calling worker.terminate() blocked for ${
+      (after - before) / 1000
+    } seconds.`,
+  );
+}
```

Comment:
This will be flaky, especially on slower user machines, but I'm pretty sure that will get flakes on CI too. Almost all time sensitive tests we had exhibited this behavior at some point (eg. sometimes runner on CI are just slow for unknown reason). 

I wonder how we could test it without being dependent on a concrete time range.